src/components/withWindowVisibility/index.desktop.js

```diff
@@ -0,0 +1,76 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import getComponentDisplayName from '../../libs/getComponentDisplayName';
+import Visibility from '../../libs/Visibility';
+import ELECTRON_EVENTS from '../../../desktop/ELECTRON_EVENTS';
+
+const windowVisibilityPropTypes = {
+    isWindowVisible: PropTypes.bool.isRequired,
+};
+
+export default function (WrappedComponent) {
+    const propTypes = {
+        forwardedRef: PropTypes.func,
+    };
+
+    const defaultProps = {
+        forwardedRef: () => {},
+    };
+
+    class WithWindowVisibility extends Component {
+        constructor(props) {
+            super(props);
+
+            this.windowFocusListener = null;
+            this.windowBlurListener = null;
+
+            this.state = {
+                isVisible: Visibility.isVisible(),
+            };
+        }
+
+        componentDidMount() {
+            this.windowFocusListener = window.electron.on(ELECTRON_EVENTS.FOCUS, () => {
+                this.setState({isVisible: Visibility.isVisible()});
+            });
+            this.windowBlurListener = window.electron.on(ELECTRON_EVENTS.BLUR, () => {
+                this.setState({isVisible: Visibility.isVisible()});
+            });
```

Comment:
Any thoughts on avoiding this platform specific HOC by doing something like `Visibility.onFocus()` ?

Reply:
Ah ok gotcha, yeah that sounds like a cleaner idea. I'll take a look.